README.md

```diff
@@ -1,16 +1,89 @@
 # Namespaced Merkle Tree (NMT)
+![Go version](https://img.shields.io/badge/go-1.14-blue.svg)
+[![API Reference](https://camo.githubusercontent.com/915b7be44ada53c290eb157634330494ebe3e30a/68747470733a2f2f676f646f632e6f72672f6769746875622e636f6d2f676f6c616e672f6764646f3f7374617475732e737667)](https://pkg.go.dev/github.com/lazyledger/nmt)
+![golangci-lint](https://github.com/lazyledger/nmt/workflows/golangci-lint/badge.svg?branch=master)
+![Go](https://github.com/lazyledger/nmt/workflows/Go/badge.svg)
+![codecov.io](https://codecov.io/github/lazyledger/nmt/coverage.svg?branch=master)
 
-This repository contains (or rather will contain) a namespaced Merkle tree library.
+
+A Namespaced Merkle Tree is
+> [...] an ordered Merkle tree that uses a modified hash function
+  so that each node in the tree includes the range of
+  namespaces of the messages in all of the descendants
+  of each node. The leafs in the tree are ordered by the
+  namespace identifiers of the messages.
+  In a namespaced Merkle tree, each non-leaf node in
+  the tree contains the lowest and highest namespace
+  identifiers found in all the leaf nodes that are descendants of the non-leaf node, in addition to the hash of
+  the concatenation of the children of the node. This
+  enables Merkle inclusion proofs to be created that prove to a verifier that all the elements of the tree for
+  a specific namespace have been included in a Merkle
+  inclusion proof.
+
+The concept was first introduced by [@musalbas] in the LazyLedger [academic paper].
```

Comment:
Missing link?

Reply:
All URLs for links are at the at the bottom of the page. Makes it easier to read the raw text. Also you have a good overview of all references at one place. Links look and work the same though (see the rendered version: https://github.com/lazyledger/nmt/blob/ismail/add_concrete_implementation/README.md)